### ISSUE TITLE
Failed sending HTTP2 data

### ISSUE THREAD

**Issue by MrksHfmn** [go-gitea/gitea#15526]
- Gitea version (or commit ref): docker image gitea:latest (Gitea version 1.15.0+dev-156-gb1e138511 built with GNU Make 4.3, go1.16.3 : bindata, timetzdata, sqlite, sqlite_unlock_notify)
- Git version: git version 2.29.2
- Operating system: Debian Buster (with backports)
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite

## Description
I am trying to push/pull code into a repo, however i always get the following error message:
`fatal: konnte nicht auf 'https://git.domain.de/user/repo.git/' zugreifen: Failed sending HTTP2 data`

The commits are not successfull.

## Docker-compose.yaml
```
---
version: '3.8'

services:
  gitea-db:
    image: mariadb
    container_name: gitea-db
    command: --transaction-isolation=READ-COMMITTED --log-bin=ROW
    restart: unless-stopped
    volumes:
      - /opt/docker/gitea/db:/var/lib/mysql
    environment:
      MYSQL_ROOT_PASSWORD_FILE: /run/secrets/gitea_db_root_password
      MYSQL_PASSWORD_FILE: /run/secrets/gitea_db_user_password
      MYSQL_DATABASE: gitea
      MYSQL_USER: gitea
      MYSQL_INITDB_SKIP_TZINFO: 1
    secrets:
      - gitea_db_root_password
      - gitea_db_user_password
    networks:
      - gitea

  gitea:
    image: gitea/gitea:latest
    container_name: gitea
    restart: unless-stopped
    volumes:
      - /opt/docker/gitea/data:/data
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
    environment:
      - USER_UID=1000
      - USER_GID=1000
      - DB_TYPE=mysql
    labels:
      - "traefik.enable=true"
      - "traefik.docker.network=traefik"
      - "traefik.http.middlewares.gitea-https.redirectscheme.scheme=https"
      - "traefik.http.routers.gitea-http.entrypoints=http"
      - "traefik.http.routers.gitea-http.rule=Host(`git.domain.de`)"
      - "traefik.http.routers.gitea-http.middlewares=gitea-https@docker"
      - "traefik.http.routers.gitea.entrypoints=https"
      - "traefik.http.routers.gitea.rule=Host(`git.domain.de`)"
      - "traefik.http.routers.gitea.tls=true"
      - "traefik.http.routers.gitea.tls.certresolver=http"
      - "traefik.http.routers.gitea.middlewares=secHeaders@file"
      - "traefik.http.services.gitea.loadbalancer.server.port=3000"
    depends_on:
      - gitea-db
    networks:
      - traefik
      - gitea

secrets:
  gitea_db_root_password:
    file: /opt/docker/_secrets/gitea_db_root_password
  gitea_db_user_password:
    file: /opt/docker/_secrets/gitea_db_user_password

networks:
  traefik:
    external: true
  gitea:
    driver: bridge
    name: gitea
```

**Comment by lunny** [comment]
Could you visit the web ui? Which protocol did you push? HTTP or SSH?

**Comment by MrksHfmn** [comment]
> 
> 
> Could you visit the web ui? Which protocol did you push? HTTP or SSH?

Web UI works without any problems. I'm pushing/pulling over https

**Comment by Rilgamon** [comment]
Same error when I try to fetch any git repo ... so this is not a gitea problem but a debian I'd guess?

```
root:/opt/gitea# git fetch
fatal: konnte nicht auf 'https://github.com/go-gitea/gitea/' zugreifen: Failed sending HTTP2 data
```

```
root@grml:/opt/pixelfed# git fetch
fatal: konnte nicht auf 'https://github.com/pixelfed/pixelfed.git/' zugreifen: Failed sending HTTP2 data
```

**Comment by MrksHfmn** [comment]
> 
> 
> Same error when I try to fetch any git repo ... so this is not a gitea problem but a debian I'd guess?
> 
> ```
> root:/opt/gitea# git fetch
> fatal: konnte nicht auf 'https://github.com/go-gitea/gitea/' zugreifen: Failed sending HTTP2 data
> ```
> 
> ```
> root@grml:/opt/pixelfed# git fetch
> fatal: konnte nicht auf 'https://github.com/pixelfed/pixelfed.git/' zugreifen: Failed sending HTTP2 data
> ```
> 
> Edit: After building own git from source its back to working.
> Edit2: Might be related to https://daniel.haxx.se/blog/2021/04/14/curl-7-76-1-h2-works-again/

Thank you for your tests! I already thought this was curl related. "Failed sending HTTP2 data" is a curl error message.
Buster Backports has curl 7.74.0. The bug seems to be fixed in 7.76.1

**Comment by Rilgamon** [comment]
My first test was going back to stable which did not work either.
The strange thing is that the "bug" was introduced with 7.76.0.
And while compiling own curl first did not work but after compiling git too it worked.
Since I've no real clue about programming and even compiling I'd guess that git from backports and stable was/is compiled with curl 7.76.0 not using libcurl while the included libcurl in stable/backports still was/is older?
